modules/cli/src/main/scala/scala/cli/commands/SetupIde.scala

```diff
@@ -37,51 +45,57 @@ object SetupIde extends ScalaCommand[SetupIdeOptions] {
       downloadDeps(inputs, options.buildOptions, options.shared.logger)
     }
 
-    val argv = {
-      val commandIndex = rawArgv.indexOf("setup-ide")
-      val withBspCommand =
-        if (commandIndex < 0) rawArgv // shouldn't happen
-        else rawArgv.take(commandIndex) ++ Array("bsp") ++ rawArgv.drop(commandIndex + 1)
-
-      // Ensure the path to the CLI is absolute
-      val progName = rawArgv(0)
-      if (progName.contains(File.pathSeparator)) {
-        val absoluteProgPath = os.FilePath(progName).resolveFrom(Os.pwd).toString
-        absoluteProgPath +: withBspCommand.drop(1)
-      }
-      else withBspCommand
-    }
+    val dir = options.bspDirectory
+      .filter(_.nonEmpty)
+      .map(os.Path(_, Os.pwd))
+      .getOrElse(inputs.workspace / ".bsp")
+
+    val bspName            = options.bspName.map(_.trim).filter(_.nonEmpty).getOrElse("scala-cli")
+    val bspJsonDestination = dir / s"$bspName.json"
+    val scalaCliBspJsonDestination = inputs.workspace / ".scala" / "scala-cli-bsp.json"
 
-    val name = options.bspName.map(_.trim).filter(_.nonEmpty).getOrElse("scala-cli")
+    // Ensure the path to the CLI is absolute
+    val progName =
+      if (rawArgv(0).contains(File.separator)) {
+        os.FilePath(rawArgv(0)).resolveFrom(Os.pwd).toString
+      }
+      else rawArgv(0)
 
     val details = new BspConnectionDetails(
-      name,
-      argv.toList.asJava,
+      bspName,
+      List(
+        progName,
+        "bsp",
+        inputs.workspace.toNIO.toAbsolutePath.toString(),
+        "--json-options",
+        scalaCliBspJsonDestination.toNIO.toAbsolutePath().toString()
```

Comment:
No need to call `.toNIO.toAbsolutePath().toString()`, `os.Path`s are already absolute (and normalized).
```suggestion
        scalaCliBspJsonDestination.toString
```